packages/core/src/useForm/useFormActions.ts

```diff
@@ -170,32 +191,143 @@ export function useFormActions<TForm extends FormObject = FormObject, TOutput ex
     }
   }
 
-  async function reset(state?: Partial<ResetState<TForm>>, opts?: SetValueOptions) {
-    if (state?.values) {
-      form.setInitialValues(state.values, opts);
-    }
+  /**
+   * Typeguard for checking if a value is a Partial<ResetState<T>>.
+   *
+   * @template T - Type parameter for ResetState
+   * @param value - Value to check
+   * @returns True if value has ResetState properties
+   */
+  function isResetState<T>(value: unknown): value is Partial<ResetState<T>> {
+    return (
+      typeof value === 'object' &&
+      value !== null &&
+      !Array.isArray(value) &&
+      ('value' in value || 'touched' in value || 'revalidate' in value)
+    );
+  }
+
+  async function reset(): Promise<void>;
+  async function reset(state: Partial<ResetState<TForm>>, opts?: SetValueOptions): Promise<void>;
+  async function reset<TPath extends Path<TForm>>(path: TPath): Promise<void>;
+  async function reset<TPath extends Path<TForm>>(
+    path: TPath,
+    state: Partial<ResetState<PathValue<TForm, TPath>>>,
+    opts?: SetValueOptions,
+  ): Promise<void>;
+
+  // Implementation signature
+  async function reset<TPath extends Path<TForm>>(
+    pathOrStateOrUndefined?: TPath | Partial<ResetState<TForm>>,
+    stateOrOptsOrUndefined?: Partial<ResetState<PathValue<TForm, TPath>>> | SetValueOptions,
+    optsOrUndefined?: SetValueOptions,
+  ): Promise<void> {
+    /**
+     * CASE 1: reset()
+     * Reset the form to its initial state.
+     */
+    if (pathOrStateOrUndefined === undefined) {
+      wasSubmitted.value = false;
+
+      form.revertValues();
+      form.revertTouched();
+      form.revertDirty();
+      submitAttemptsCount.value = 0;
+      isSubmitAttempted.value = false;
 
-    if (state?.touched) {
-      form.setInitialTouched(state.touched, opts);
+      await validate();
+
+      return Promise.resolve();
     }
 
-    wasSubmitted.value = false;
+    /**
+     * CASE 2: reset(state, opts?)
+     * Reset the form to a specific state.
+     */
+    if (isResetState<TForm>(pathOrStateOrUndefined)) {
```

Comment:
We can tell if this is CASE 2 or the path cases if the `pathOrStateOrUndefined` is a string which is more reliable check than object properties.